ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -287,10 +287,10 @@ static VirtualizationRootHandle FindUnusedIndexOrGrow_Locked()
     if (RootHandle_None == rootIndex)
     {
         // No space, resize array
-        uint16_t newLength = MIN(s_maxVirtualizationRoots * 2u, INT16_MAX + 1u);
-        if (newLength <= s_maxVirtualizationRoots)
+        uint16_t newLength;
+        if (__builtin_mul_overflow(s_maxVirtualizationRoots, 2, &newLength))
```

Comment:
@pmj I think this is equivalent but please confirm your intent with the old code.

Reply:
Not quite, the handles are signed integers hence the limit of `INT16_MAX + 1u` (= 32768), whereas the max value of `newLength` is `UINT16_MAX` (= 65535).